fluentdhec/lambda_function.py

```diff
@@ -34,94 +40,102 @@ def lambda_handler(event, context):
+def parse_raw_event(log: Dict) -> Dict:
     return {
         "sourcetype": "generic:k8s",
         "index": os.environ['SPLUNK_INDEX'],
         "event": log['message'],
     }
 
 
-def parse_hsm_log_event(log):
+def parse_hsm_log_event(log: Dict) -> Dict:
     return {
         "sourcetype": "cloudhsm",
         "index": os.environ['SPLUNK_INDEX'],
         "event": hsmdecoder.jsoniser(log['message'])
     }
 
 
-def send_to_hec(payload):
+def send_to_hec(payload: str) -> None:
     hec = PyHEC(os.environ['SPLUNK_HEC_TOKEN'], os.environ['SPLUNK_HEC_URL'])
     hec.send(payload)
 
 
-def extract_time(message):
-    # this matches date and 2 / 4 digit year values (19 vs 2019)
-    # rDY could be either at start or end (2019 MM DD or DD MM 2019)
-    rDY = r"\d{2,4}"
-    # rMon could be character (May) or 2 digit (05)
-    rMon = r"(?:[a-zA-Z]+|\d\d)"
-    # for optional timezones
-    rTimeZone = r"[\+-]\d\d\:?\d\d"
-    # rTime matches timestamps with optional seconds and optional timezones
-    rTime = rf"\d\d\:\d\d(?:\:\d\d)?(?:[\.,]\d+)? ?(?:Z|AM|PM|{rTimeZone})?"
-    regex = rf'(?P<d>{rDY}.{rMon}.{rDY}).(?P<t>{rTime})|usecs:(?P<s>[0-9]+)'
-    matches = re.search(regex, message)
-    res = False
-    # if there are no matches, return false
+def extract_time(message: str) -> int:
+    """
+    Parses a message to return a timestamp in seconds.
+
+    >>> extract_time("usecs:12345")
+    12345
+    >>> extract_time("2009-Feb-13 23:31:30")
+    1234567890
+    """
+    day_year = r"\d{2,4}"
+    month = r"(?:[a-zA-Z]+|\d\d)"
+    timezone = r"[\+-]\d\d\:?\d\d"
+    time = rf"\d\d\:\d\d(?:\:\d\d)?(?:[\.,]\d+)? ?(?:Z|AM|PM|{timezone})?"
+    time_matcher = (rf"(?P<date>{day_year}.{month}.{day_year})."
+                    rf"(?P<time>{time})|usecs:(?P<timestamp>[0-9]+)")
+    matches = re.search(time_matcher, message)
     if not matches:
-        return res
-    # if there's a `usecs` match, use this
-    if matches.group("s"):
-        res = int(matches.group("s"))
-    # use the date and time matches separately to make a string
-    # the dateparser library is particular about certain characters
-    elif matches.group("d") and matches.group("t"):
-        try:
-            date = matches.group("d")
-            time = matches.group("t").replace(",", ".")
-            dt = dateparser.parse(f"{date} {time}")
-            res = int(dt.timestamp())
-        except AttributeError:
-            res = False
-    return res
+        raise ValueError("No recognisable timestamp in message")
+    if matches.group("timestamp"):
+        timestamp_seconds = int(matches.group("timestamp"))
+    elif matches.group("date") and matches.group("time"):
+        date = matches.group("date")
+        time = matches.group("time").replace(",", ".")
+        datetime = dateparser.parse(f"{date} {time}")
+        timestamp_seconds = int(datetime.timestamp())
+    else:
+        raise ValueError("No recognisable timestamp in message")
+    return timestamp_seconds
```

Comment:
@akinnane dude, you're gonna love what we've done to this function

Reply:
:fire: Function now only returns `int`s, doesn't return `false` on errors, now uses exceptions
:fire: Variable names now human readable
:fire: Constant comments are not needed, due to good variable names
:fire: Added type hinting
:fire: docstring includes doctest'd examples